services/report/__init__.py

```diff
@@ -1239,3 +1255,31 @@ def save_parallel_report_to_archive(
             "chunks_path": chunks_url,
             "files_and_sessions_path": files_and_sessions_url,
         }
+
+
+@sentry_sdk.trace
+def delete_uploads_by_sessionid(upload: Upload, session_ids: list[int]):
+    """
+    This deletes all the `Upload` records corresponding to the given `session_ids`.
+    """
+    db_session = upload.get_db_session()
+    uploads = (
+        db_session.query(Upload.id_)
+        .filter(Upload.report == upload.report, Upload.order_number.in_(session_ids))
```

Comment:
What's order_number? Don't think I've ever run into this one 😅

Reply:
It is the `session_id`, so the incremental ID within the `report_json`.